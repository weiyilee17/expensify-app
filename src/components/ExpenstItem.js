import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseItem = ({ id, description, amount, createdAt }) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h3>Description: {description}</h3>
		</Link>
		<p>Amount: {amount} - createdAt: {createdAt}</p>
	</div>
);

export default ExpenseItem;