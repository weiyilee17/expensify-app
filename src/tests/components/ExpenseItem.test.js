import React from 'react';
import { shallow } from 'enzyme';

import ExpenstItem from '../../components/ExpenstItem';
import testExpenses from '../fixtures/expenses';

test('should render first fixture expense', () => {
  const wrapper = shallow(<ExpenstItem {...testExpenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
