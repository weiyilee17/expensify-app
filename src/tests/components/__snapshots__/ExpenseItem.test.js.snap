// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render first fixture expense 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Description: 
      Gum
    </h3>
  </Link>
  <p>
    Amount: 
    195
     - createdAt: 
    0
  </p>
</div>
`;
